pkg/applier/applier_err.go

```diff
@@ -73,11 +73,13 @@ func SkipErrorForResource(err error, id core.ID, strategy actuation.ActuationStr
 
 // largeResourceGroupError indicates that the source repo has too many objects
 // to manage with a single resource group.
-func largeResourceGroupError(err error, id core.ID) status.Error {
-	e := fmt.Errorf("source repository contains too many resource objects to inventory with a single ResourceGroup: "+
-		"inventory apply failed: %v: %s. "+
-		"To mitigate, spread the source objects across multiple repositories. "+
-		"For how, see https://cloud.google.com/anthos-config-management/docs/how-to/breaking-up-repo",
-		id, err)
+func largeResourceGroupError(_ error, id core.ID) status.Error {
+	// the error is not used because it's confusing to the user.  The current core error
```

Comment:
This seems like an anti-pattern to completely swallow the real error and replace it with a different error. The `misleading error` is coming from the apiserver and is the same thing a user would see if they tried to apply the oversized CR with `kubectl`. If the error is misleading then I think the action item is to improve the error string upstream in apiserver

Reply:
I agree that working with etcd and kubernetes layers is probably the right course of actions here, however it seems like that kind of a change is many months away and wouldn't serve the customers that have current versions of Kubernetes for some time.

Also agreed that generally swallowing errors is discouraged but I think the UX here is what error our user sees so we should probably figure out what leads to the least amount of support load.  As it stands today this is one of the most common scenarios so we should evaluate the potential remediations.

Reply:
I'm not totally convinced this is an improvement over the current behavior. I think I'd be more open to it the underlying error is still logged but not surfaced on the RSync status, but even then it still seems like a bit of a code smell